### ISSUE TITLE
Pushers that take spesific items

### ISSUE THREAD

**Issue by piecubed** [joe7575/techpack#5]
What if there was a pusher that you could click on and it would have slots thatnyou could put items in, and the items in the slots were the only items it would push?  I also wonder anout a stackwise pusher... and a stackwise distributer...  they would push/distribute items in stacks, onstead of by single items.

**Comment by joe7575** [comment]
Thanks for your response. To understand it better, I have the following questions:
to 1) What is the use case for the "configurable" pusher? I see no big difference to the distributor then.
to 2) Yes possible, but it was a decision at the very beginning for single items instead of stacks. Do you need this high throughput?

**Comment by piecubed** [comment]
Also, the stackwise pushers are an idea, you have to rewrite alot...

**Comment by joe7575** [comment]
Again, I see no benefit with "configurable" pusher. You could do the same with Distributors. Or in other words, if "configurable" pusher would be available, Distributors no longer would be necessary, right?

**Comment by joe7575** [comment]
There are no nodes, which could produce more than a fast pusher could handle. So why even faster pushers? I still see no use case.
Only to have more expensive alternatives is no reason for further nodes/more code/complexity and effort to maintain...

**Comment by piecubed** [comment]
Yeah, i was dropping the faster pusher idea.  But think about this, i have a massive gravel sieve machine, so it produces alot of sand.  The way i have it set up, the distributer sends six sand per tick, but the six slots of nodes to be distributed, always will fill up.  There is plenty of room on the other end... but it always fills up.  So say instead if that distributer, i made it go though a gold chest... with two configurable pushers.  The first configurable pusher goes to a gold chest.  The other pushes everything else to be smelted.  But first, it would push into a distributerthar separates the smeltabe items from the non smeltables.  And, it would be nice if there was swo sets of slots, one of items to push, and one of items not to push. So then if i had a chest full of whatever, and i wanted everything but this... so i put that into the not slots, and then it grabs everything else.

**Comment by piecubed** [comment]
And distributers are still needed, unless you come ip with sorting tubes, and juction tubes.

**Comment by joe7575** [comment]
I've played around with the idea of a "sorting pusher", but the tubelib API does not allow such kind of node. I won't change/extend the API because more and more mods add support for tubelib.
To implemlent that, the "sorting" feature has to come from the chest not the pusher.
One idea would be to have a sorting chest with several inventories, one on each side. But this is to similar to the distributor.
Another idea is a sorting/pushing chest which will push stacks instead of items. These pushing chests can be arranged to a ring working as as tubelib backbone or alternatively as a warehouse.

**Comment by joe7575** [comment]
I've added 3 further nodes as "tubelib_addons3". Not on GitHub so far but under test on my own server. Will come soon...

These node are high performance nodes allowing to push stacks instead of items:
- a HighPerf Chest 
- a HighPerf Distributor
- a HighPerf Pushing Chest (a combination of sorting/pushing node with inventory)
